01_Exercises/main.py

```diff
@@ -0,0 +1,74 @@
+# LAB 1
+# Calculati suma a N numere naturale.
+def sum_numbers(*numbers):
+    sum = 0
+    find = 0
+
+    for check in numbers:
+        if check < 0:
+            find = -1
+
+    for num in numbers:
+        sum = sum + num
+
+    if find < 0:
+        print("Choose only natural numbers")
+    else:
+        return sum
+
+
+print(sum_numbers(-2, 3))
+
+
+# Verificați daca un număr citit de la tastatura este prim.
+def only_prime_numbers(number):
+    divisors = []
+
+    for i in range(1, number + 1):
+        if number % i == 0:
+            divisors.append(i)
+
+    if number == 0 or number == 1:
+        print("This is NOT a prime number")
+    elif len(divisors) > 2:
+        print("This is NOT a prime number")
+    else:
+        print("This is a prime number")
+
+
+only_prime_numbers(13)
+
+
+# Calculați cel mai mare divizor comun a doua numere.
+def highest_divisors(a, b):
+    divisors_a = []
+    divisors_b = []
+    highest_divisor = []
+
+    for i in range(1, a):
+        if a % i == 0:
+            divisors_a.append(i)
+
+    for y in range(1, b):
+        if b % y == 0:
+            divisors_b.append(y)
+
+    while True:
```

Comment:
Doar o obeservatie,  `while True` cu `break` la final nu face absolut nimic. 
Algortimul de aici in colo e gresit din 2 pcte de vedere. Primul e ca in momentul in care faci `compare = compare + 1` vei apela apoi divisors_b[compare] . Problema e ca  divisors_b s-ar putea sa fie prea mica ca sa apelezi indexul deci programul o sa-ti crape. Incearca `highest_divisors(5, 10)` sa vezi ca crapa.
A doua problema e ca tu parcurgi cea mai mare lista de divizori si ii compari pe rand, asta n-o sa mearga oricum pentru ca divizorul comun s-ar putea sa fie in lista dupa un divizor care nu este comun.
Exemplu:
10, 15
10 are divizorii [1, 2, 5]
15 are divizorii [1, 3, 5]
Programul tau va compara in felul urmator:
1 == 1? da merg mai de parte in amandoua listele si adaug 1 ca divizor comun
2 == 3? nu merge mai departe in prima lista si la a doua raman pe loc
5 == 3? nu
deci o sa mi dea cel mai mare divizor comun 1, pentru ca nu a ajuns niciodata la 5

Bun, astea fiind spuse e foarte bun approachul de a calcula divizorii ambelor numere si a incerca sa gasesti cel mai mare numar comun din ambele liste. Cum faci asta cel mai simplu e cu 2 foruri:

```
max = -99999
for el1 in list1:
    for el2 in list2:
         if el1 == el2 and el1 > max:
            max = el1
```

Reply:
Intradevar acel While nu facea nimic. Cat despre restul codului stiam ca-mi trebuie un nested loop dar m-am complicat.
Insa codul asta cred ca are mai mult sens acum:

`    for a in divisors_a:
        for b in divisors_b:
            if a == b:
                highest_divisor.append(a)
            else:
                pass

    return highest_divisor[-1]`

Intrebare de ce ai mai compari max cu el1, nu cred ca face nimic.